File: geomajas/face/geomajas-face-gwt/geomajas-gwt-client/src/main/java/org/geomajas/gwt/client/Geomajas.java
/*
 * This file is part of Geomajas, a component framework for building
 * rich Internet applications (RIA) with sophisticated capabilities for the
 * display, analysis and management of geographic information.
 * It is a building block that allows developers to add maps
 * and other geographic data capabilities to their web applications.
 *
 * Copyright 2008-2010 Geosparc, http://www.geosparc.com, Belgium
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package org.geomajas.gwt.client;

import java.util.HashMap;
import java.util.Map;

/**
 * <p>
 * Global settings for the Geomajas GWT face.
 * </p>
 * 
 * @author Pieter De Graef
 */
public final class Geomajas {

	private Geomajas() {
	}

	/** Returns the current version of Geomajas as a string. */
	public static String getVersion() {
		return "1.5.4";
	}

	/**
	 * Returns a list of locales that can be used in this version of Geomajas. The default is english, and 'native'
	 * means that your browsers locale should be used (if supported - default otherwise).
	 */
	public static Map<String, String> getSupportedLocales() {
		Map<String, String> locales = new HashMap<String, String>();
		locales.put("default", "English");
		locales.put("nl", "Nederlands");
		return locales;
	}

	/**
	 * Return the base directory for the web application.
	 * 
	 * @return
	 */
	public static native String getIsomorphicDir()
	/*-{
		return $wnd.isomorphicDir;
	}-*/;
}

File: geomajas/face/geomajas-face-dojo/geomajas-dojo-client/src/main/resources/geomajas/geomajas.js
/*
 * This file is part of Geomajas, a component framework for building
 * rich Internet applications (RIA) with sophisticated capabilities for the
 * display, analysis and management of geographic information.
 * It is a building block that allows developers to add maps
 * and other geographic data capabilities to their web applications.
 *
 * Copyright 2008-2010 Geosparc, http://www.geosparc.com, Belgium
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

dojo.provide("geomajas.geomajas");
dojo.require("dojox.collections");
dojo.require("dojo.parser");
dojo.require("dijit.layout.ContentPane");
dojo.require("dijit.layout.LayoutContainer");
dojo.require("dijit.layout.SplitContainer");
dojo.require("dijit.layout.BorderContainer");
dojo.require("dijit.layout.AccordionContainer");
dojo.require("dojox.layout.FloatingPane");
dojo.require("dijit.TitlePane");
dojo.require("dojox.layout.ExpandoPane");

dojo.require("geomajas.log.log4javascript");
dojo.require("geomajas._base");
dojo.require("geomajas.config.common");
dojo.require("geomajas.event.common");
dojo.require("geomajas.spatial.common");
dojo.require("geomajas.util.common");
dojo.require("geomajas.map.common");
dojo.require("geomajas.gfx.common");
dojo.require("geomajas.action.common");
dojo.require("geomajas.controller.common");
dojo.require("geomajas.io.common");
dojo.require("geomajas.widget.attributes.common");

dojo.require("geomajas.widget.ActivityDiv");
dojo.require("geomajas.widget.DynamicToolbar");
dojo.require("geomajas.widget.ExtendedFeatureListTable");
dojo.require("geomajas.widget.FeatureDetailEditor");
dojo.require("geomajas.widget.FeatureDetailTable");
dojo.require("geomajas.widget.FeatureEditDialog");
dojo.require("geomajas.widget.FeatureListTable");
dojo.require("geomajas.widget.FloatingPane");
dojo.require("geomajas.widget.FontStyleWidget");
dojo.require("geomajas.widget.LayerTree");
dojo.require("geomajas.widget.LegendWidget");
dojo.require("geomajas.widget.LineStringStyleWidget");
dojo.require("geomajas.widget.MapWidget");
dojo.require("geomajas.widget.PolygonStyleWidget");
dojo.require("geomajas.widget.ProgressDialog");
dojo.require("geomajas.widget.ScaleSelect");
dojo.require("geomajas.widget.SearchTable");
dojo.require("geomajas.widget.TemplatePrintWidget");
dojo.require("geomajas.widget.DefaultPrintWidget");
dojo.require("geomajas.widget.TextBalloon");
dojo.require("geomajas.widget.ZoomSlider");

dojo.require("geomajas.widget.experimental.LoadingScreen");
dojo.require("geomajas.widget.experimental.Profiler");
dojo.require("geomajas.widget.experimental.DownloadDialog");
